patches/boringssl/0003-Disable-Werror.patch

```diff
@@ -0,0 +1,25 @@
+From 8cbf2c2c15c86b5619a6ea5d032838b9e9df2025 Mon Sep 17 00:00:00 2001
+From: Ivan Afonichev <iafonichev@griddynamics.com>
+Date: Tue, 4 May 2021 22:51:26 +0400
+Subject: [PATCH] Disable Werror
+
+---
+ CMakeLists.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f58e853cd..b1fec9616 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -117,7 +117,7 @@ endif()
+ if(CMAKE_COMPILER_IS_GNUCXX OR CLANG)
+   # Note clang-cl is odd and sets both CLANG and MSVC. We base our configuration
+   # primarily on our normal Clang one.
+-  set(C_CXX_FLAGS "-Werror -Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -Wvla")
++  set(C_CXX_FLAGS "-Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -Wvla")
```

Comment:
Which warnings remain without this patch, i.e. which warnings cause the compilation to fail with `-Werror`? If possible, I would prefer it if boringssl would compile without modified `CFLAGS`. Also: If we disable `-Werror` anyhow, there is no need to fix the `-Wstringop-overflow` warning (which is what the second patch does) or is there?

Reply:
Only sha256.c warnings remains after 2 google CLs. See details in upstream bug: https://bugs.chromium.org/p/boringssl/issues/detail?id=402#c4
We apply 2 google patches to address the issues and apply our workaround patch to silence warnings of remaining issues

Reply:
Ah! In that case `-Wno-error=stringop-overflow` should be sufficient and preferable over disabling `-Werror` entirely.

Reply:
Ping, would be nice to include this as part of the upcoming 31.0.2 release (see #29) :)